### PR TITLE
[3.11] gh-103559: Update bundled pip version to 23.1.1 (gh-103560)

### PR DESCRIPTION
**Pull request by pfmoore**
* Update bundled pip version to 23.1
* Update to pip 23.1.1

(cherry picked from commit b2862950dc05d313b228f02a8efdd8c9c59a8ff4)

* Issue: gh-103559

### CODE REVIEW

**Comment by pfmoore** [comment]
@pablogsal I assume there's no issue backporting this?

**Review by preveen-stack — APPROVED** [review]

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

**Comment by pfmoore** [comment]
Thanks for the approvals. This is only really waiting on core dev approval, and I'm not actually sure I need that, as it feels like it's uncontroversial enough that I can just merge my own PR.

**Pull request merged**
